python/python_frontend/fusion_record.h

```diff
@@ -1699,6 +1697,82 @@ struct ReductionOpRecord : RecordFunctor {
   PrimDataType dtype_;
 };
 
+struct ScanOpRecord : RecordFunctor {
+  ScanOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      std::string _name,
+      serde::RecordType record_type,
+      std::function<TensorView*(TensorView*, int64_t)> fusion_op,
+      int64_t dim,
+      BinaryOpType op_type)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            _name,
+            record_type),
+        fusion_op_(std::move(fusion_op)),
+        dim_(dim),
+        op_type_(op_type) {}
+  ~ScanOpRecord() override = default;
+  RecordFunctor* clone() final {
+    return new ScanOpRecord(*this);
+  }
+
+  //! Child specific hash function in lower 32 bits.
+  //! | 7 --- 4 | 3 --- 0 |
+  //! | op_type | dim     |
+  size_t hash() const final {
+    auto result = RecordFunctor::hash();
+    result |= ((static_cast<size_t>(op_type_) & 0xf) << 4);
+    return result | (static_cast<size_t>(dim_) & 0xf);
+  }
+
+  bool operator==(const RecordFunctor& other) const final {
+    auto result = false;
+    if (auto child_ptr = dynamic_cast<const ScanOpRecord*>(&other)) {
+      result = RecordFunctor::operator==(other);
+      if (result) {
+        result = result &&
+            (fusion_op_.target_type() == child_ptr->fusion_op_.target_type());
```

Comment:
For my own curiosity: since fusion_op_ is not a template and we are already comparing `target<...>()` down, I don't think we need to check target_type here.